packages/date-picker/test/overlay.test.js

```diff
@@ -17,303 +17,394 @@ function waitUntilScrolledTo(overlay, date, callback) {
-        var initialDate = new Date(2000, 1, 1);
-        overlay.scrollToDate(initialDate);
-        var closeSpy = sinon.spy();
-        overlay.addEventListener('close', closeSpy);
-        // const scrollSpy = sinon.spy(overlay, 'scrollToDate');
+        describe('date limits', () => {
+          let todayMidnight, yesterdayMidnight, tomorrowMidnight;
+
+          beforeEach(() => {
+            const today = new Date();
+            todayMidnight = new Date(0, 0);
+            todayMidnight.setFullYear(today.getFullYear());
+            todayMidnight.setMonth(today.getMonth());
+            todayMidnight.setDate(today.getDate());
+            yesterdayMidnight = new Date(todayMidnight.getTime());
+            yesterdayMidnight.setDate(todayMidnight.getDate() - 1);
+            tomorrowMidnight = new Date(todayMidnight.getTime());
+            tomorrowMidnight.setDate(todayMidnight.getDate() + 1);
+            overlay.minDate = null;
+            overlay.maxDate = null;
+          });
 
-        overlay.$.todayButton.disabled = true;
+          it('should not be disabled by default', () => {
+            expect(overlay.$.todayButton.disabled).to.be.false;
+          });
 
-        waitUntilScrolledTo(overlay, initialDate, () => {
-          const lastScrollPos = overlay.$.monthScroller.position;
-          tap(overlay.$.todayButton);
+          it('should not be disabled if today is inside the limits', () => {
+            overlay.minDate = yesterdayMidnight;
+            overlay.maxDate = tomorrowMidnight;
+            expect(overlay.$.todayButton.disabled).to.be.false;
+          });
 
-          expect(overlay.$.monthScroller.position).to.equal(lastScrollPos);
-          // FIXME: fails in FF + Polymer 1.x
-          // expect(scrollSpy.called).to.be.false;
-          expect(closeSpy.called).to.be.false;
-          done();
+          it('should not be disabled if today is min', () => {
+            overlay.minDate = todayMidnight;
+            expect(overlay.$.todayButton.disabled).to.be.false;
+          });
+
+          it('should not be disabled if today is max', () => {
+            overlay.maxDate = todayMidnight;
+            expect(overlay.$.todayButton.disabled).to.be.false;
+          });
+
+          it('should be disabled if the limits are in past', () => {
+            overlay.maxDate = yesterdayMidnight;
+            expect(overlay.$.todayButton.disabled).to.be.true;
+          });
+
+          it('should be disabled if the limits are in future', () => {
+            overlay.minDate = tomorrowMidnight;
+            expect(overlay.$.todayButton.disabled).to.be.true;
+          });
         });
       });
+    });
+  });
 
-      describe('date limits', () => {
-        let todayMidnight, yesterdayMidnight, tomorrowMidnight;
+  describe('scrollToDate', () => {
+    beforeEach(async () => {
+      overlay = fixtureSync(`
+        <vaadin-date-picker-overlay-content
+          style="position: absolute; top: 0;"
```

Comment:
nit: this shouldn't be necessary and you can probably remove it from all the fixtures in this test suite.

The reason for this CSS was that early version of `web-test-runner` was printing test results in the browser (rather than in the console) - that caused content to shift, so I had to use this workaround 🙈

Reply:
Done.